### ISSUE TITLE
Performance issue of Future.delayed hack

### ISSUE THREAD

**Issue by ranquild** [brianegan/dart_redux_epics#14]
This line:
https://github.com/brianegan/dart_redux_epics/blob/master/lib/src/epic_middleware.dart#L50

runs timer on every action dispatched within store. In our app it adds visible performance bottleneck. Can you please add option to dispatch actions immediately (can be done backward-compatible)? We could use this option and do not use `async*` actions instead.

We are still using 0.7.x version and we will use it for few months from now (dart 2.0 migration). Can you also please publish this option for latest 0.7.x package as well?

**Comment by brianegan** [comment]
Sure, now that Dart2 is supporting this use-case a bit better, I'll see if I can actually just remove it.

**Comment by ranquild** [comment]
@brianegan It seems that I've found easy solution to this issue without deleting hack completly:

```
if (_flag) {
      _actions.add(action);
    } else {
      new Future.delayed(Duration.ZERO, () {
        _actions.add(action);
        _flag = true;
      });
    }
```

Tested it in my view and it seems to work, however there are failing tests in this library because action can be dispatched sooner than expected.

Can you please look at this?

About perf tests, I will try extract code to simulate problem. Problem is that someone is dispatching actions to store on mouse move or similar.

**Comment by brianegan** [comment]
Thanks @ranquild -- hrm, this is an interesting problem overall. When I removed the `Future.delayed` hack, I notice other things break as well, such as the `TypedEpic` helpers, so I'm actually not sure a simple `_flag` solution will quite work.

I'm trying to see if I can rewrite the middleware a bit to support all of these use-cases without any hacks that are causing trouble.

**Comment by brianegan** [comment]
Ok, I've published `0.10.2` & `0.7.1`, which includes an option to disable support for `async*` functions until Dart 2 properly supports this use case via https://github.com/dart-lang/sdk/issues/33818

To disable support for async* functions, you can create your EpicMiddleware like this:

```dart
new EpicMiddleware<String>(
  myEpic,
  supportAsyncGenerators: false,
)
```
